### ISSUE TITLE
Some numbers eg-30 crashes the app

### ISSUE THREAD

**Issue by anshumanv** [anshumanv/evolutions-webapp#29]
Find a workaround to handle such cases.

- [x] I ensure that this issue is not a duplicate of any current open issues(s).
- [ ] Assign this issue to me (I want to work on this).
- [ ] I will send my PR only after I'm assigned to this issue by any maintainer.
- [ ] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

**Comment by avsingh999** [comment]
@anshumanv   name also crashes the app eg-porygon

**Comment by anshumanv** [comment]
Yeah thanks, I'll fix that in some time.

**Comment by deadbirdi** [comment]
I'll take a look at this one.

**Comment by deadbirdi** [comment]
So, from what I can tell, it seems like the`react-poke-sprites` package is what's causing the errors. For example, when you user input `porygon`, sprites knows what porygon is, but it doesnt know what `porygonz` is, and thats where the error happens.

It looks like `30` crashes the app because it's trying to look for a pokemon called `nidoranf` instead of `nidoran`? 

Here's an error log:

```
Error: Invalid pokemon name "porygonz" 2 react-poke-sprites.js:2026
```
@anshumanv I believe `react-poke-sprites` is your package? I think this is just an adjustment of naming conventions that's necessary.

**Comment by anshumanv** [comment]
react-poke-sprites uses https://www.npmjs.com/package/pokemon-gif ro resolve name/number and return gif and for obtaining evolution chain we are using a json file obtained from that pokemon simulation game so that causes naming conflicts, we can do some error handling so display no results and not break the app. What do you think we should do?

**Comment by anshumanv** [comment]
There must be many naming conflicts which we don't know of now.

**Comment by deadbirdi** [comment]
Writing a unit test that goes through each name and digit is probably a good idea

—
Zachary E Sohovich
Developer at Isolary
zach@isolary.com (mailto:zach@isolary.com)

**Comment by anshumanv** [comment]
Makes sense yeah.

**Comment by anshumanv** [comment]
I'm not particularly good with writing tests, can you do it once you've got sufficient time at hand?

**Comment by deadbirdi** [comment]
I've never done one either, but I would love to learn! Do we already have a test platform? If not, I've been itching to work with Jest (which plays nice with `create-react-app`)

Per the comment about what I think we should do, I think we shouldn't display nothing when the user enters a correct Pokemon name. I think we need to modify the JSON file to match the naming convention.

**Comment by anshumanv** [comment]
> I've never done one either, but I would love to learn! Do we already have a test platform? If not, I've been itching to work with Jest (which plays nice with create-react-app)

Awesome! :smile: 

Yeah, that certainly is doable but then it's some ~800 entries, which will require some time, I'll try to get someone who is willing to do this. Yeah infact let's go by updating the JSON.

**Comment by dericcain** [comment]
@anshumanv @sneakycrow Jest has a new feature that can iterate through the `pokedex` and test each entry. This way you do not have to manually write a test for each one of them. Here are the docs to that specific Jest feature:

https://jestjs.io/docs/en/api#testeachtable-name-fn-timeout
